### PR TITLE
Various PAL60 related fixes and improvements.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
This is related to https://github.com/dolphin-emu/dolphin/pull/2520.

The main thing this PR does is combine the "Graphics -> Advanced -> Enable Progressive Scan" and "Config -> Wii -> Enable PAL60" options into a single set of radio buttons that more closely resembles the way you'd set video modes on actual console.

There's a ~~few~~ side effect~~s~~ of this:

* The previously selectable combination of "PAL60 off, Progressive on" is no longer possible. Since this combination is not actually selectable on a console, I think this is actually an improvement.
* ~~Since the Progressive Scan option is now in the Wii config tab, it no longer applies to Gamecube games. The only thing the option did for Gamecube games before is set if the emulated Gamecube had a Component Cable connected to it, which would allow games to display the prompt about switching into Progressive Scan if you booted a supported game while holding the B button. This procedure will now just be always available.~~

Additionally, I have added the PAL60 option to the Movie header, as well as allowing it to be overriden by GameINIs. This fixes [issue 7847](https://code.google.com/p/dolphin-emu/issues/detail?id=7847).

Lastly, I have changed the boot procedure to automatically disable PAL60 if an NTSC game boots. This fixes issues such as [7714](https://code.google.com/p/dolphin-emu/issues/detail?id=7714) and [8036](https://code.google.com/p/dolphin-emu/issues/detail?id=8036).

### CODE REVIEW

**Comment by mimimi085181** [comment]
I haven't tested this yet or looked at the code, but the description is wrong. On console it is not possible to select pal60 and progressive at the same time. You mean that on console it's not possible to enable a 50Hz progressive mode, which was possible until now in Dolphin. Pal60 on or off was applied to progressive mode, switching between 50 and 60Hz progressive mode, but progressive mode on console is always 60Hz.

**Comment by JosJuice** [comment]
@mimimi085181 It is possible to select PAL60 (IPL.E60) and progressive (IPL.PGS) at the same time. It's what happens when you select 480p on a PAL console.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by mimimi085181** [comment]
Hmm, ok, sorry. Since you can choose between pal50, pal60 and progressive mode, and there are bits for pal60 and progressive mode, i was expecting that progressive mode does not set the pal60 bit.

File: Source/Core/DolphinWX/Config/WiiConfigPane.cpp [hunk 1]
```diff
@@ -39,29 +40,33 @@ void WiiConfigPane::InitializeGUI()
 	m_system_language_strings.Add(_("Traditional Chinese"));
 	m_system_language_strings.Add(_("Korean"));
 
+	m_video_mode_strings.Add(_("576i"));
+	m_video_mode_strings.Add(_("480i"));
+	m_video_mode_strings.Add(_("480p"));
```
Comment (JosJuice) [inline Source/Core/DolphinWX/Config/WiiConfigPane.cpp]: It would be useful to note the framerate here, because users might not know what the correlation between resolution and framerate is. `576i (50 Hz)` / `480i (60 Hz)` / `480p (60 Hz)`

**Comment by AdmiralCurtiss** [comment]
Changed the option labels to indicate framerate.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinWX/Config/WiiConfigPane.cpp [hunk 2]
```diff
@@ -83,8 +88,15 @@ void WiiConfigPane::InitializeGUI()
 
 void WiiConfigPane::LoadGUIValues()
 {
+	if (SConfig::GetInstance().m_LocalCoreStartupParameter.bProgressive) {
```
Comment (BhaaLseN) [inline Source/Core/DolphinWX/Config/WiiConfigPane.cpp]: Either put the braces on a new line, or omit them since its just a single-line statement

Comment (AdmiralCurtiss) [inline Source/Core/DolphinWX/Config/WiiConfigPane.cpp]: Ha, sorry. I'm used to same-line braces.

**Comment by skidau** [comment]
While the Progressive Scan and the PAL60 options result in a similar effect, they are two different options.  The PAL60 option is a shortcut so that the user does not need to boot up the System Menu to change the PAL60 option.  The Progressive Scan option is the emulation of having a physical composite or component cable plugged into the console.  On the real PAL Wii, the EDTV option ( http://i.imgur.com/Xtt7YJa.jpg ) inside the System Menu is enabled or disabled by presence of a component cable and is automatically detected by the console.  This PR removes the ability to emulate the component cable configuration.

In summary, combining the two options (essentially removing the progressive scan option) is not correct.

**Comment by BhaaLseN** [comment]
I don't think its a big loss if we cannot emulate the "without component cable" case; or is there something we'd miss out in that case (except a disabled option on the Wii Menu)?

**Comment by lioncash** [comment]
I'd rather err on the side of correctness here. This is an emulator of course, so it makes sense to support it if the system is capable of it.

**Comment by BhaaLseN** [comment]
So, a "Component Cable" checkbox then? :3

**Comment by skidau** [comment]
It is possible to write Wii and GC software to detect the component cable (as the SysMenu does).  Imagine a homebrew author who like to test their software on consoles which are connected via a composite cable.  They wouldn't be able to if this change were added.

**Comment by lioncash** [comment]
@BhaaLseN Wasn't the progressive scan checkbox the "Component Cable" checkbox? At least that's what I understand from what @skidau has said.

**Comment by BhaaLseN** [comment]
Yeah, except that it seems to have been used for 2 distinct things (unless I'm mistaken). The question is whether an end-user recognized what the setting actually did (and re-wording might be appropriate).

**Comment by AdmiralCurtiss** [comment]
It was called "Enable Progressive Scan" and set the IPL.PGS option of the Wii SYSCONF, which is not the same as having a component cable connected. Although, yes, it did also also affect that in Dolphin.

**Comment by mimimi085181** [comment]
I'm confused, i can select select all 3 video modes on my pal wii, when i have my component cable connected.

**Comment by AdmiralCurtiss** [comment]
Right, there's nothing stopping you on real console from connecting a Component cable but still setting the output to Interlaced. A "Component Cable" checkbox that is separate from the Progressive Scan would be appropriate if we want to be entirely correct.

**Comment by skidau** [comment]
Hardcoding m_DTVStatus.component_plugged to 1 is where this PR removes the option.

Updating IPL.PGS is simulating what the System Menu does to tell the Wii game that it is running on a progressive scan cable system.

**Comment by AdmiralCurtiss** [comment]
No, IPL.PGS is not the same as having a Component cable connected. If you plug in a Component cable but set the option [here](http://i.imgur.com/0y3DLI1.png) to 576i or 480i, the console sets IPL.PGS to 0.

**Comment by AdmiralCurtiss** [comment]
But you're right that this removes the option. I can re-add it with a separate checkbox no problem.

**Comment by skidau** [comment]
I didn't say that IPL.PGS is the same as having a Component Cable connected.  IPL.PGS is software and a Component Cable is hardware.

**Comment by AdmiralCurtiss** [comment]
So what do you want me to do here now?

**Comment by JosJuice** [comment]
In my opinion, there should be a component cable checkbox where the progressive scan checkbox was before.

**Comment by AdmiralCurtiss** [comment]
Seems reasonable to me. Should the Component Cable being disabled also disable the 480p option like it does on console?

**Comment by skidau** [comment]
Please remove the change from the Source/Core/Core/HW/VideoInterface.cpp file.  This is where the PR has affected "hardware" emulation.  The other parts of the PR are SysMenu software emulation areas.

**Comment by AdmiralCurtiss** [comment]
Removed.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
Disabling the 480p option and automatically changing it to 480i if it already was selected would be nice if it isn't too hard to do, but I don't care too much about it.

LGTM, regardless of whether the component cable option makes it in. The branch needs to be rebased, though.

**Comment by BhaaLseN** [comment]
I'm not so sure if we should do it that way round. If we decide to disable anything, we should disable the component cable option if 480p is selected (and provide an appropriate tooltip on why its disabled; even tho this seems a bit backwards from a technical standpoint). Otherwise end-users might not know theres a 3rd option in the list (or why they cannot select a 3rd option if its not hidden away) if this depends on a "component cable" checkbox that the majority wouldn't care about.

Or we could just opt to override the users decision if they decide to select 480p with component cable off, set it to on at run-time and just show an OSD message telling them that their settings are inconsistent and we chose to enable component cable for them (that way we can leave the GUI alone to do just the bare minimum in the context of this PR).

**Comment by JosJuice** [comment]
Fair point - we should definitely make this as easy as possible for users, and the component cable option is not the important one. Both of your suggestions sound fine, except I don't want there to be yet another OSD message cluttering things down.

**Comment by BhaaLseN** [comment]
I can live with no OSD message at all; since those who might care about the option are few and will probably hit the channel (or bug tracker) anyways if they don't immediately see why 480p works even though they disabled cc.

**Comment by JosJuice** [comment]
One caveat, though: If the component cable option is set at runtime when 480p is set, it should only be done for Wii games, because GameCube games don't care about the Wii 480p setting, and GameCube games is where the component cable setting matters the most.

**Comment by AdmiralCurtiss** [comment]
Rebased.

**Comment by AdmiralCurtiss** [comment]
Should we decide on the potential Component Cable option here or should that be a different PR?

**Comment by BhaaLseN** [comment]
@dolphin-emu-bot rebuild

I think it makes sense to do it in another PR, unless you can whip up something in a few minutes :)

**Comment by AdmiralCurtiss** [comment]
I'm fine with doing that in a different PR.

**Comment by JosJuice** [comment]
I also think it fits best in another PR, so that the important stuff can be merged more quickly.

File: Source/Core/Core/BootManager.cpp [hunk 3]
```diff
@@ -168,6 +170,12 @@ bool BootCore(const std::string& _rFilename)
 		VideoBackend::ActivateBackend(StartUp.m_strVideoBackend);
 		core_section->Get("GPUDeterminismMode", &StartUp.m_strGPUDeterminismMode, StartUp.m_strGPUDeterminismMode);
 
+		// Some NTSC games crash if the PAL60 option is enabled
```
Comment (Buddybenj) [inline Source/Core/Core/BootManager.cpp]: For future people reading the code, add some examples (e.g. Doc Louis's Punch-Out!!, 1942).

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JMC47** [comment]
I like this change a lot, tested it, and it makes PAL60 so much better.

Weird and maybe unrelated, but a lot of PAL NES games stutter at 50 FPS audio wise, but won't run at 60 FPS even with PAL60 set.  Maybe another PAL timing problem?

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by AdmiralCurtiss** [comment]
Nintendo was really lazy with the PAL Wii Virtual Console, so nearly all PAL VC releases run in 50Hz only like they did back on actual NES/SNES/etc. PAL hardware. As far as I know the only exceptions are games that were not released in Europe before, and TG16 games.

**Comment by AdmiralCurtiss** [comment]
Y'know, with all the rebasing I have to do here, I think I'm gonna split this into separate PRs.

**Comment by degasus** [comment]
Sorry for the latest merge conflict :/ It is because or PR #2594, maybe this helps you with rebasing :/
